rust/src/proofs/api.rs

```diff
@@ -421,16 +421,17 @@ pub unsafe extern "C" fn fil_unseal_range(
 
         info!("unseal_range: start");
 
+        use filepath::FilePath;
         use std::os::unix::io::{FromRawFd, IntoRawFd};
 
-        let mut sealed_sector = std::fs::File::from_raw_fd(sealed_sector_fd_raw);
-        let mut unseal_output = std::fs::File::from_raw_fd(unseal_output_fd_raw);
+        let sealed_sector = std::fs::File::from_raw_fd(sealed_sector_fd_raw);
+        let unseal_output = std::fs::File::from_raw_fd(unseal_output_fd_raw);
 
-        let result = filecoin_proofs_api::seal::unseal_range(
+        let result = filecoin_proofs_api::seal::get_unsealed_range(
             registered_proof.into(),
             c_str_to_pbuf(cache_dir_path),
-            &mut sealed_sector,
-            &mut unseal_output,
+            sealed_sector.path().unwrap(),
+            unseal_output.path().unwrap(),
```

Comment:
unseal_output is always an anonymous pipe -> https://github.com/filecoin-project/lotus/blob/master/extern/sector-storage/ffiwrapper/sealer_cgo.go#L323

We do that because lotus-miner wants to have the same data format for sectors passed into PreCommit, and unsealed data, so unsealed sector files look like
```
[32GiB of fr32 padded data][RLE+ of allocated byte ranges][4B int of that RLE+ length]
```

Reply:
Ok, I can probably leave that argument alone then.  Thanks

Reply:
An alternative is to just have a new API call, or a versioned call for unseal_range or something

Reply:
It might be easiest to have a new call with slightly different args, and I think we can hide this new call at the FFI layer, so I don't think anything would change from the caller side.  I'll see if there's a way around that, but not sure at the moment.

Reply:
Basically, I want this to be public, so we can use that: https://github.com/filecoin-project/rust-fil-proofs/blob/fix-ram-unseal/filecoin-proofs/src/api/mod.rs#L261

Reply:
Or some small wrapper around it, like the other unseal calls now are

Reply:
Hah, nevermind, we already have what I want (as of this PR) -- just need to plumb it through: https://github.com/filecoin-project/rust-fil-proofs/blob/fix-ram-unseal/filecoin-proofs/src/api/mod.rs#L195